deserializer.go

```diff
@@ -19,50 +22,104 @@ func newDeserializer() *deserializer {
 	}
 }
 
-func (d *deserializer) newHostsToTargets() map[string]*Targets {
+func (d *deserializer) newRulesets() (map[string]*ruleset, *radix.Tree, error) {
 	start := time.Now()
-	data := MustAsset("targets.gob")
+	data := MustAsset("rulesets.gob")
 	buf := bytes.NewBuffer(data)
 
 	dec := gob.NewDecoder(buf)
-	hostsToTargets := make(map[string]*Targets)
-	err := dec.Decode(&hostsToTargets)
+	rulesets := make([]*Ruleset, 0)
+	err := dec.Decode(&rulesets)
 	if err != nil {
 		d.log.Errorf("Could not decode: %v", err)
-		return nil
+		return nil, nil, err
 	}
-	d.log.Debugf("Loaded HTTPS Everywhere in %v", time.Now().Sub(start).String())
 
 	// The compiled regular expressions aren't serialized, so we have to manually
 	// compile them.
-	for _, v := range hostsToTargets {
-		for _, r := range v.Rules.Rules {
-			r.from, _ = regexp.Compile(r.From)
-		}
+	plains := make(map[string]*ruleset)
+	wildcards := radix.New()
+	for _, rs := range rulesets {
+		wildcards = d.addRuleset(rs, plains, wildcards)
```

Comment:
Now that you're not using the immutable version, you don't need to keep returning the wildcards tree.